sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala

```diff
@@ -192,8 +192,10 @@ class AnalysisErrorSuite extends AnalysisTest {
 
   errorTest(
     "DISTINCT and FILTER cannot be used in aggregate functions at the same time",
-    CatalystSqlParser.parsePlan("SELECT count(DISTINCT a) FILTER (WHERE c > 1) FROM TaBlE2"),
-    "DISTINCT and FILTER cannot be used in aggregate functions at the same time" :: Nil)
+    CatalystSqlParser.parsePlan("SELECT sum(DISTINCT b), " +
+      "count(DISTINCT a) FILTER (WHERE c > 1) FROM TaBlE2"),
+    "When there are multiple DISTINCT aggregate expressions acting on different " +
+    "fields, any DISTINCT aggregate expression not allow use FILTER clause." :: Nil)
```

Comment:
Can you add some tests for the foldable case?

Reply:
OK. I will add tests in group-by-filter.sql

Reply:
You've already added the tests? (I couldn't find them..)

Reply:
At
https://github.com/apache/spark/blob/45d69714a7a1bb6dc8a05a11a2b885174f5d3fca/sql/core/src/test/resources/sql-tests/inputs/group-by-filter.sql#L67

Reply:
`emp_name` is foldable? I meant a query like `select key, sum(distinct value), avg(distinct 1) ...`